internal/provider/provider.go

```diff
@@ -211,15 +216,17 @@ func buildClient(p *schema.Provider, b *authentication.Builder, partnerId string
 		TerraformVersion: p.TerraformVersion,
 	}
 
-	client, err := clientBuilder.Build(p.StopContext())
-	if err != nil {
-		return nil, err
+	stopCtx, ok := schema.StopContext(ctx) //nolint:SA1019
+	if !ok {
+		stopCtx = ctx
 	}
 
-	// replaces the context between tests
-	p.MetaReset = func() error { //nolint unparam
-		client.StopContext = p.StopContext()
-		return nil
+	client, err := clientBuilder.Build(stopCtx)
+	if err != nil {
+		return nil, diag.Diagnostics{diag.Diagnostic{
```

Comment:
And i think there is a helper function in the SDK so this can. be one line?

Reply:
Yes indeed!